### PR TITLE
adding more information about Interfaces in Java

### PR DESCRIPTION
**Pull request by renetanikolova**
- [ ] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #XXXXX

### CODE REVIEW

**Comment by RandellDawson** [comment]
Closed and Reopened this PR to attempt to resolve a specific Travis build failure.

File: guide/english/java/data-abstraction/index.md [hunk 1]
```diff
@@ -9,6 +9,16 @@ Likewise in Object-oriented programming, abstraction is a process of hiding the
 
 In Java, abstraction is achieved using Abstract classes and interfaces.
 
+Interfaces were introduced in Java to enhance Java’s single-inheritance model, multiple inheritance created too many problems for programmers and compiler writers. 
+All methods in interface are public, we may not write it every time, like all of them are abstract, but we never write the key word abstract.
```
Comment (Manish-Giri) [inline guide/english/java/data-abstraction/index.md]:
```suggestion
All methods in an Interface are implicitly `public` so we need not write the `public` keyword each time. 
```

Java 8 introduced default methods, which can contain full implementation, so "all of them are abstract" doesn't really hold true anymore.

File: guide/english/java/data-abstraction/index.md [hunk 2]
```diff
@@ -9,6 +9,16 @@ Likewise in Object-oriented programming, abstraction is a process of hiding the
...
+The interface should have at least one method but not too many.
```
Comment (Manish-Giri) [inline guide/english/java/data-abstraction/index.md]: Is there a reference that you can link to which supports this statement?

File: guide/english/java/data-abstraction/index.md [hunk 3]
```diff
@@ -9,6 +9,16 @@ Likewise in Object-oriented programming, abstraction is a process of hiding the
...
+
+Advantages using interfaces:
+It decouples your code.
```
Comment (Manish-Giri) [inline guide/english/java/data-abstraction/index.md]: Please use list formatting to list out the advantages. Please review the [style guide for guide articles](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/docs/style-guide-for-guide-articles.md) if you need help with formatting.

**Review by Manish-Giri — CHANGES_REQUESTED** [review]
Hi @renetanikolova 

Thanks for creating this PR. Please address the suggested changes.

**Review by Manish-Giri — COMMENTED** [review]
Hi,

As there has been no response from your end for a while, I am closing this PR.

Feel free to re-open it if you wish to apply the suggested changes.

Happy Coding!

**Pull request closed without merging**